pkg/controller/jobframework/validation.go

```diff
@@ -73,6 +85,12 @@ func ValidateUpdateForQueueName(oldJob, newJob GenericJob) field.ErrorList {
 	if !newJob.IsSuspended() {
 		allErrs = append(allErrs, apivalidation.ValidateImmutableField(QueueName(oldJob), QueueName(newJob), queueNameLabelPath)...)
 	}
+
+	oldHasPrebuiltWl, oldWlName := prebuiltWorkload(oldJob)
+	newHasPrebuiltWl, newWlName := prebuiltWorkload(newJob)
+	if oldHasPrebuiltWl || newHasPrebuiltWl {
```

Comment:
This check does not seem necessary, here we do an analogous check: https://github.com/kubernetes-sigs/kueue/blob/615dc8e42d160174962f379dd8767e95298a1457/pkg/controller/jobframework/validation.go#L80